### ISSUE TITLE
Allow SQL queries to include a GEOJSON polygon or multipolygon

### ISSUE THREAD

**Issue by benlaken** [Vizzuality/sql2gee#7]
currently, aggregations to vectors is done by passing a kwag (geojson), but this should be extended to accept the geojson from within a postgis like function.

**Comment by benlaken** [comment]
Generic SQL requests containing embedded geojson will be made as follows:

```sql
SELECT ST_SUMMARYSTATS() FROM "srtm90_v4" WHERE ST_INTERSECTS(ST_SetSRID(ST_GeomFromGeoJSON('{"type":"Polygon","coordinates":[[[-43.39599609375,-4.740675384778361],[-43.39599609375,-4.959615024698014],[-43.17626953125,-4.806364708499984],[-43.39599609375,-4.740675384778361]]]}'), 4326), the_geom)
```

Will need to recognise the use of the keywords as meaning make an aggregation over the geojson contained within.

Note, we are using the ST_INTERSECTS in a different way than its postgis counterpart (which simply meant return a boolean to indicate if a pair of vectors intersects): in our case, they mean use the following provided geojson as an area over which to aggregate.

**Comment by benlaken** [comment]
Problem. After passing this long statement to the SQL2GEE class, the sqlparse tokeniser lib is unable to recognise the dataset/table Identifier. Unsure why this is. But it seems work must be done on the target_data() property method to resolve this.

**Comment by benlaken** [comment]
Seems that this bug can be resolved by altering the target_data() function to look at the length of the item token, instead of the type:
Changing: 
`if isinstance(item, Identifier):`
to
`if len(item.value.strip()) > 0:`

Change made in f80274fda1

**Comment by benlaken** [comment]
### Problems

* POSTGIS syntax not part of sqlparse ability to tokenise - will have to work out something custom. This is a problem with the nested and complex nature of the function. (Probably will need to use a parser?)

* ST_INTERESECTS and accompanying nested functions/values are prefaced with a WHERE statement, so will need to distinguish this from regular where cases.
* Unsure how a statement including a geojson and a logical where would look/function.
    e.g. from data where geojson and where data > x

**Comment by benlaken** [comment]
Solved with 4291743e2293b91bd64a3f491d8f48182c1eba47
